src/commands/release-asset/download.ts

```diff
@@ -0,0 +1,145 @@
+/**
+ * @license
+ * Copyright 2025 Balena Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Command, Flags } from '@oclif/core';
+import { getBalenaSdk, stripIndent, getVisuals } from '../../utils/lazy';
+import { commitOrIdArg } from '../release';
+import * as fs from 'fs';
+import * as path from 'path';
+import { pipeline } from 'stream/promises';
+import { Transform } from 'stream';
+
+export default class ReleaseAssetDownloadCmd extends Command {
+	public static description = stripIndent`
+		Download a release asset.
+
+		Download a release asset with the specified key. By default, the file will be saved
+		with the original filename. Use the --output flag to specify a different output path.
+	`;
+
+	public static examples = [
+		'$ balena release-asset download 1234567 --key config.json',
+		'$ balena release-asset download a777f7345fe3d655c1c981aa642e5555 --key app.tar.gz --output ./downloads/app.tar.gz',
+		'$ balena release-asset download 1234567 --key large-file.bin -o output.bin',
+	];
+
+	public static args = {
+		commitOrId: commitOrIdArg({
+			description: 'the commit or ID of the release',
+			required: true,
+		}),
+	};
+
+	public static flags = {
+		key: Flags.string({
+			description: 'the key of the release asset to download',
+			required: true,
+		}),
+		output: Flags.string({
+			char: 'o',
+			description: 'output path for the downloaded file',
+		}),
```

Comment:
This combines changing download directory and renaming the file. Perhaps we should keep them separate? If I just wanted to change the file name, I would be a bit annoyed that I need to go get the full path I want to send the file to just to download with a new name

Reply:
But looking at line 89 I see it could indeed be used to only rename. Unclear to me at first, but I'm not sure if that's just me